frontend/src/pages/warehouse/ModalAddWarehouse.js

```diff
@@ -0,0 +1,324 @@
+                                validationErrors: [],
+                                serverErrors: ''
+                            }));
+                            setDispatchersList([]);
+                            setDropdownMenuName("select type");
+                            props.onChange(false, warehouseDto)
+                        }
+                    }
+                )
+        }
+    };
+
+    const showDispatchers = dispatchersList.map(disp =>
+        <div key={disp.id}>
+            {disp.name + " " + disp.surname + ", username: " + disp.username}
+            <FaTrash style={{color: '#1A7FA8', textAlign: 'center'}}
+                     onClick={() => {
+
+                         setDispatchersList(
+                             dispatchersList.filter((dispatcher) => dispatcher.id !== disp.id));
+                     }}
+            />
+        </div>
+    );
+
+    return (
+        <>
+            <Modal
+                show={props.props}
+                backdrop="static"
+                onHide={() => {
+                    setErrors({
+                        validationErrors: [],
+                        serverErrors: ''
+                    });
+                    props.onChange(false);
+                    setDropdownMenuName("select type");
+                }}
+                aria-labelledby="modal-warehouse"
+                className="shadow"
+                centered
+            >
+                <Modal.Header closeButton>
+                    <Modal.Title id="modal-warehouse">
+                        Add warehouse
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    {errors.serverErrors && <ErrorMessage message={errors.serverErrors}/>}
+                    <Form>
+                        <Form.Group controlId="identifier" style={{padding: '5px 10px'}}>
+                            Identifier
+                            <Form.Control type="text"
+                                          onChange={handleIdentifier}
+                                          className={errors.validationErrors.includes("identifier")
+                                              ? "form-control is-invalid" : "form-control"}/>
+                            <Form.Control.Feedback type="invalid">
+                                Please provide a valid identifier.
+                            </Form.Control.Feedback>
+                        </Form.Group>
+
+                        <Form.Group controlId="type" style={{padding: '5px 10px'}}>
+                            <Dropdown>
+                                <Dropdown.Toggle variant="btn btn-outline-primary" id="dropdown-basic">
+                                    {dropdownMenuName}
+                                </Dropdown.Toggle>
+                                <Dropdown.Menu>
+                                    <Dropdown.Item onClick={() => handleType("FACTORY")}>FACTORY</Dropdown.Item>
```

Comment:
There are 3 special functions in js: bind, call, apply. Here we may get benefit by usage of bind function. It would be very informative and helpful for you to read about all of them and then apply bind function here.